# Group Project: Emission data and plots

Sub-Question: How are the emissions related to the average travel distance of people in the Netherlands?

In [1]:
import pandas as pd
import plotly.express as px
import plotly.subplots as sp


df_road = pd.read_csv('Road emissions.csv', delimiter=',')
df_road


Periods  Carbon dioxide (CO2)  Carbon monoxide (CO)  PM10 (fine dust)
0     2005                 30200                 393.4               8.6
1     2010                 29600                 370.4               6.3
2     2015                 28100                 317.0               4.5
3     2016                 28900                 307.2               4.4
4     2017                 29200                 308.1               4.3
5     2018                 29400                 286.1               4.2
6     2019                 29300                 268.4               4.0
7     2020                 26100                 224.1               3.5
8     2021                 26600                 217.3               3.6

The dataset used for emissions was provided on cbs.nl, featuring emissions measured on Dutch territory in million kg of CO2, CO and PM10. The data was collected on various transport modes in the Netherlands spanning from 2005 until 2021, with a yearly interval from 2015. The values applicable to the research objective include total road traffic emissions, since the focus of the objective is on vehicles on Dutch roads. The dataset was last updated in September of 2022.

## Carbon dioxide:

In [32]:
# The dataset is imported
df_road = pd.read_csv('Road emissions.csv', delimiter=',')
df_road

# Outlier periods are removed
df_roadlimited = df_road.drop(0)
df_roadlimited = df_roadlimited.drop(1)
df_roadlimited

# Line chart

figco2line = px.line(df_roadlimited, x='Periods', y='Carbon dioxide (CO2)',
                 title='Total road traffic emissions (Million kg)')
figco2line.update_traces(line_color='red')
figco2line.show()


# Bar chart

figco2bar = px.bar(df_roadlimited, x='Periods', y='Carbon dioxide (CO2)',
                 title='Total road traffic emissions (Million kg)')
figco2bar.update_yaxes(range=(25000,31000))
figco2bar.update_traces(marker_color='red')
figco2bar.show()






The CO2 emission data was plotted as shown in Figure \ref{fig:co2} in the form of a line and bar chart. CO2 was chosen from the data since it is the most commonly used measure when discussing climate change and human pollution effects on the environment. There is a clear drop in CO2 emissions from 2019 to 2020.

## Carbon monoxide:

In [33]:
# Line chart

figcoline = px.line(df_roadlimited, x='Periods', y='Carbon monoxide (CO)',
                 title='Total road traffic emissions (Million kg)')
figcoline.update_traces(line_color='green')
figcoline.show()

# Bar chart

figcobar = px.bar(df_roadlimited, x='Periods', y='Carbon monoxide (CO)',
                 title='Total road traffic emissions (Million kg)')
figcobar.update_yaxes(range=(200,400))
figcobar.update_traces(marker_color='green')
figcobar.show()

## PM10 (Fine Dust):

In [36]:
# Line chart

figpm10line = px.line(df_roadlimited, x='Periods', y='PM10 (fine dust)',
                 title='Total road traffic emissions (Million kg)')
figpm10line.update_traces(line_color='blue')
figpm10line.show()

# Bar chart

figpm10bar = px.bar(df_roadlimited, x='Periods', y='PM10 (fine dust)',
                 title='Total road traffic emissions (Million kg)')
figpm10bar.update_yaxes(range=(3,5))
figpm10bar.update_traces(marker_color='blue')
figpm10bar.show()

## Combined:

In [5]:

# For as many traces that exist per Express figure, get the traces from each plot and store them in an array.
# This is essentially breaking down the Express fig into it's traces
figco2line_traces = []
figco2bar_traces = []
for trace in range(len(figco2line["data"])):
    figco2line_traces.append(figco2line["data"][trace])
for trace in range(len(figco2bar["data"])):
    figco2bar_traces.append(figco2bar["data"][trace])

#Create a 1x2 subplot
this_figure = sp.make_subplots(rows=1, cols=2) 

# Get the Express fig broken down as traces and add the traces to the proper plot within in the subplot
for traces in figco2line_traces:
    this_figure.append_trace(traces, row=1, col=1)
for traces in figco2bar_traces:
    this_figure.append_trace(traces, row=1, col=2)

# this_figure.add_annotation(text="CO2 (Million kg)")
this_figure.update_layout(title_text="Total road traffic emissions (Million kg)")
this_figure.show()




Combination with Simmon's data

In [30]:
df_use = pd.read_csv('Road usage.csv', delimiter=',')
df_use

Period  Average number of ride per day  Average km travelled per ride  \
0    2017                            2.81                          10.38   
1    2018                            2.89                          10.37   
2    2019                            2.82                          10.44   
3    2020                            2.34                           8.77   

   Average distance travelled per day  
0                             29.1678  
1                             29.9693  
2                             29.4408  
3                             20.5218

In [29]:
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
from plotly.subplots import make_subplots

fig_com = make_subplots(specs=[[{"secondary_y": True}]])
fig_com.add_trace(
    go.Bar(x = df_roadlimited["Periods"], y=df_roadlimited["Carbon dioxide (CO2)"], name="Yearly CO2 Emissions", marker_color = 'blue'),
    secondary_y=False)
fig_com.add_trace(
    go.Scatter(x=df_use["Period"], y=df_use["Average distance travelled per day"], name="Average distance travelled per day"),
    secondary_y=True)
fig_com.update_layout(yaxis1 = dict(range=[25000,30000]))
fig_com.update_layout(title_text = "Distance travelled compared to CO2 emissions")
fig_com.update_layout(yaxis1_title="Million Kilograms")
fig_com.update_layout(yaxis2_title="Kilometers")
fig_com.update_layout(xaxis_title="Year")
fig_com

When overlaying the data of yearly CO2 emissions with the average distance travelled on Dutch roads per person per day, it is clear that there is an observable correlation between the two trends. This is to be expected, since a decrease in kilometers travelled must mean a lower vehicle activity and thus less engine operation times, which results in less emissions from the combustion of the vehicles engines. This is portrayed in the graph above, where there is a significant drop from 29.4 km to 20.5 km travelled per person per day between 2019 and 2020. This corresponds to a drop in emissions in the same time frame 29.3 million tons to 26.1 million tons of CO2.